### DISCUSSION TITLE
Implement Mock KV in-memory

### DISCUSSION THREAD

**Discussion by bogdanbiv** [skymethod/denoflare#60] [Ideas]
When I've run denoflare serve on my worker I noticed Key-Value store is not implemented locally:
```
Error servicing request Error: KVNamespaceRpcStub.list not implemented.
    at RpcKVNamespace.list (blob:null/78de0f04-7a96-42c8-bc1a-2e61ce2dbfef:225:15)
```

I would need a mock KV for my tests so I'd like to try porting  the mock implementation from edge-mock:
https://github.com/samuelcolvin/edge-mock/blob/11e6c03e3649d9c84cf1f1fad7bdabe9f86f9d18/src/kv_namespace.ts#L34

Does this sound like a good idea? Would this mock KV find an audience with the rest of Denoflare users?

My current idea is that the mock KV would be implemented as a part of a global this to be able to retain a reference to it between requests. Or maybe I could make it a part of some persistent Denoflare instance? For now I know I need to reuse typedefs  `KVNamespace, KVListResult, KVListResultKey` from `common/cloudflare_workers_types.d.ts`

Please do comment.

**Comment by johnspurlock-skymethod** [comment]
This would be a great addition to denoflare proper, similar to the existing local memory R2 bucket implementation [InMemoryR2Bucket](https://github.com/skymethod/denoflare/blob/c01be53b43ad1d55986922e159e75971bbccffe4/cli/in_memory_r2_bucket.ts#L4) which can be configured in local development with an R2 binding of `local:memory:<bucket-name>`

If you submit a PR with a new class `InMemoryKvNamespace implements KVNamespace` in pure TS with no dependencies, I'll wire it up to the local dev server so everyone can use it
